### DISCUSSION TITLE
why the emit method is so slowly

### DISCUSSION THREAD

**Discussion by ZhuqiushiCasual** [miguelgrinberg/python-socketio#1294] [Q&A]
When I interact with the frontend or test my API using Postman, there's an approximate 5-second difference between the time the client receives the 'emit' event and the time I log as the sending time on the server side.

This is my server python code
```python
    await sio.emit('newMessage', answer_result.to_json(), to=sid, namespace=default_namespace)
    time_emit = datetime.now().strftime("%H:%M:%S")
    logger.info(f"newMessage emit time{time_emit}")
# The 'sio' in the above code comes from following code self._sio
    def __init__(
            self,
            app: FastAPI,
            mount_location: str = "/ws",
            socketio_path: str = "socket.io",
            cors_allowed_origins: Union[str, list] = '*',
            async_mode: str = "asgi",
            **kwargs
    ) -> None:
        # TODO: Change Cors policy based on fastapi cors Middleware
        self._sio = AsyncServer(async_mode=async_mode, cors_allowed_origins=cors_allowed_origins, ping_interval=120,
                                ping_timeout=120,
                                **kwargs)
        self._app = ASGIApp(
            socketio_server=self._sio, socketio_path=socketio_path
        )

        app.mount(mount_location, self._app)
        app.sio = self._sio
```

this is server emit time
![image](https://github.com/miguelgrinberg/python-socketio/assets/75727106/250e9aa2-82da-461a-8f45-7fa3b32aec2c)
this is client receive emit time
![image](https://github.com/miguelgrinberg/python-socketio/assets/75727106/111ff46c-57fd-48ef-80c9-a1a0c35b9ee8)

my pip repository version
```
fastapi                0.103.1
python-socketio        5.5.0
```
If you could take some time to look into this, I would greatly appreciate it.

just now, I found that when adding an emit behind the emit in the above code, the first emit will be sent to the client immediately.
Is there a queue here? How to turn off the queue? Or increase the speed of the queue sending?

**Comment by miguelgrinberg** [comment]
I don't see anything that can cause a delay in the code that you shared. My guess is that your FastAPI application blocks, preventing the background work that python-socketio does to send a message from happening in a timely manner.
